src/problems/ParticleSF/test_particle_SF.cpp

```
@@ -0,0 +1,203 @@
+	const amrex::Real exp_num_low_mass_stars_total = prob_star_formation * nx * ny * nz;
+	const amrex::Real exp_num_high_mass_stars_total = prob_star_formation * exp_num_high_mass_stars_percell * nx * ny * nz;
+
+	// get total mass of the final particles
+	const auto [real_data_final, idata_final] =
+	    sim.particleRegister_.getParticleDescriptor(quokka::ParticleType::StochasticStellarPop)->getParticleDataAtLevel(0);
+	const int mass_idx = 3;
+	double high_mass_stars_total_mass = 0.0;
+	double all_stars_total_mass = 0.0;
+	int num_high_mass_stars = 0;
+	int num_low_mass_stars = 0;
+	const int num_all_stars = static_cast<int>(real_data_final.size());
+	for (int i = 0; i < num_all_stars; ++i) {
+		if (idata_final[i][0] != static_cast<int>(quokka::StellarEvolutionStage::LowMassComposite)) {
+			high_mass_stars_total_mass += real_data_final[i][mass_idx];
+			num_high_mass_stars++;
+		} else {
+			num_low_mass_stars++;
+		}
+		all_stars_total_mass += real_data_final[i][mass_idx];
+	}
+	const double mean_mass_high_mass_stars = high_mass_stars_total_mass / num_high_mass_stars;
+	const double mean_mass_high_mass_stars_Msun = mean_mass_high_mass_stars / M_sol;
+	amrex::Print() << "Probability of star formation in every cell = " << prob_star_formation << "\n";
+
+	amrex::Print() << "Total particle mass          = " << all_stars_total_mass / M_sol << " Msun\n";
+	amrex::Print() << "Total Number of high mass stars    = " << num_high_mass_stars << "\n";
+	amrex::Print() << "Total Number of low mass stars     = " << num_low_mass_stars << "\n";
+	amrex::Print() << "Total Number of all stars          = " << num_all_stars << "\n";
+	amrex::Print() << "Mean mass of high mass stars = " << mean_mass_high_mass_stars_Msun << " Msun\n";
+
+	// relative error
+	const double sigma_high_mass_stars = std::sqrt(exp_num_high_mass_stars_total);
+	const double rel_error_Mstar_high_mean = std::abs(mean_mass_high_mass_stars_Msun - exp_Mstar_high_mean) / sigma_high_mass_stars;
```

The calculation of sigma_high_mass_stars is not correct. The sigma of Mstar_high_mean should have the same dimension as Mstar_high_mean, i.e. M_sol. The calculation of the sigma of the mean of a power-law distribution is not trivial, so you could simply report the relative error and make sure it's visually small enough.